### PR TITLE
#24071 Fixing: Render Page with the dotContentMap Object

### LINKED ISSUE #24071: Render Page with specific variant is not working with SYSTEM_CONTAINER
### Problem Statement

When you try to render a Page using that use the SYSTEM_CONTAINER in a specific Variant is it render with the DEFAULT Variant.

### Steps to Reproduce

- Create A Page and use the SYSTEM_TEMPLATE.
- Add A Generic Content into this Page with a title of "DEFULT VERSION"
- Create an Experiment in this page and Crate a variant.
- Edit the Variant and edit the Contentlet change the title to "EXPERIMENT VERSION"
- Close the Dialog and you are not going to see the change.
- Even if you refresh the page is render with the DEFAULT version.

### Acceptance Criteria

- The page must be render with the specific variant when we are using the SYSTEM Container

### dotCMS Version

last in master

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* Use a fallback by Variant and Language 

https://github.com/dotCMS/core/pull/24209/files#diff-8120c6eddd966c9d6be3ee4dbc5af4a94bea9946b0627eb49991dde286610e49R129

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java [hunk 1]
```diff
@@ -1949,7 +1954,11 @@ public void injectJSCodeWithoutHead() throws WebAssetException, DotDataException
     /**
      * Method to test: {@link HTMLPageAssetRenderedAPI#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
      * When: Try to render a page with a specific {@link Variant}} and a specific {@link Language}
+<<<<<<< HEAD
+     * and the page had a Contentlet that has a version in the variant but not in the DEFAULT Variant
+=======
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java]: fix this conflicts

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java [hunk 2]
```diff
@@ -2008,6 +2017,119 @@ public void renderPageWithNoDefaultContentlet() throws WebAssetException, DotDat
...
+                        .setPageMode(PageMode.LIVE)
+                        .build(),
+                mockRequest, mockResponse);
+
+        Assert.assertTrue(html.contains("VARIANT" + language.getId()));
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPI#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
+     * When: Try to render a page with a specific {@link Variant}} using a {@link Container} thar is using the dotContentMap
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java:2071]: fix typo

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java [hunk 3]
```diff
@@ -2008,6 +2017,119 @@ public void renderPageWithNoDefaultContentlet() throws WebAssetException, DotDat
...
+                        .build(),
+                mockRequest, mockResponse);
+
+        Assert.assertTrue(html.contains("<div>VARIANT" + language.getId()+ "</div>"));
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPI#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
+=======
+>>>>>>> origin/master
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java]: fix conflict

**Review by victoralfaro-dotcms — APPROVED** [review]

**Review by dsilvam — CHANGES_REQUESTED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/24209/commits/7135f6d5cf901210a453eecb927ce359aa8d7dc2#diff-aac9fd429a1e8f9ab8efad97d4259f8be79b41078a9c0750704c907760f2d85cL1957

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/24209/files#diff-aac9fd429a1e8f9ab8efad97d4259f8be79b41078a9c0750704c907760f2d85cR2071

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegrationTest.java]: done https://github.com/dotCMS/core/pull/24209/commits/254e7f7241a60ccf01f48988672b46fc94dd1eed#diff-aac9fd429a1e8f9ab8efad97d4259f8be79b41078a9c0750704c907760f2d85cL2127

File: dotCMS/src/main/java/com/dotcms/variant/business/web/VariantWebAPI.java [hunk 4]
```diff
@@ -28,6 +29,9 @@ public interface VariantWebAPI {
     RenderContext getRenderContext(final long tryingLang, final String identifier,
             final PageMode pageMode, final User user);
 
+    ContentletVersionInfo getContentletVersionInfoByFallback(final long tryingLang, final String identifier,
```
[the remark this answers is not in the document]
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/variant/business/web/VariantWebAPI.java:68]: done https://github.com/dotCMS/core/pull/24209/files#diff-45dbdd64c70c9bbcc5509ea0609065408ee20034d258b0ffa416e73c6976f0f5R68

**Review by dsilvam — APPROVED** [review]

**Pull request merged**